crates/router/src/core/tokenization.rs

```diff
@@ -0,0 +1,185 @@
+        created_at: common_utils::date_time::now(),
+        updated_at: common_utils::date_time::now(),
+        flag: enums::TokenizationFlag::Enabled,
+        version: enums::ApiVersion::V2,
+    };
+
+    // Insert into database
+    let tokenization = db
+        .insert_tokenization(
+            tokenization_new,
+            &(merchant_key_store.clone()),
+            key_manager_state,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to insert tokenization record")?;
+
+    // Convert to TokenizationResponse
+    Ok(hyperswitch_domain_models::api::ApplicationResponse::Json(
+        api_models::tokenization::TokenizationResponse {
+            id: tokenization.id,
+            created_at: tokenization.created_at,
+            flag: tokenization.flag,
+        },
+    ))
+}
+
+#[instrument(skip_all)]
+#[cfg(all(feature = "v2", feature = "tokenization_v2"))]
+pub async fn get_token_vault_core(
+    state: SessionState,
+    merchant_account: &domain::MerchantAccount,
+    merchant_key_store: &domain::MerchantKeyStore,
+    query_params: (id_type::GlobalTokenId, bool),
+) -> RouterResponse<serde_json::Value> {
+    let db = state.store.as_ref();
+    let key_manager_state = &(&state).into();
+    // Get the tokenization record from the database
+    let tokenization = db
+        .get_entity_id_vault_id_by_token_id(
+            &query_params.0,
+            &(merchant_key_store.clone()),
+            key_manager_state,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to get tokenization record")?;
+
+    if tokenization.flag == enums::TokenizationFlag::Disabled {
+        return Err(errors::ApiErrorResponse::InvalidRequestData {
+            message: "Tokenization is disabled".to_string(),
+        }
+        .into());
+    }
+
+    let vault_request = pm_types::VaultRetrieveRequest {
+        entity_id: tokenization.merchant_id.clone(),
+        vault_id: hyperswitch_domain_models::payment_methods::VaultId::generate(
+            tokenization.locker_id.clone(),
+        ),
+    };
+
+    let vault_data = pm_vault::retrive_value_from_vault(&state, vault_request)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to retrieve vault data")?;
+
+    let data_json = vault_data
+        .get("data")
+        .cloned()
+        .unwrap_or(serde_json::Value::Null);
+
+    // Mask sensitive data if needed
+    let response_data = if !query_params.1 {
+        mask_sensitive_data(data_json)
+    } else {
+        data_json
+    };
+
+    // Create the response
+    Ok(hyperswitch_domain_models::api::ApplicationResponse::Json(
+        response_data,
+    ))
+}
+
+#[cfg(all(feature = "v2", feature = "tokenization_v2"))]
+fn mask_sensitive_data(value: serde_json::Value) -> serde_json::Value {
```

Comment:
why we are using masking like this? there is secret which does the same right?

Reply:
We can introduce a masking strategy which does the same and using masking itself. So that there will be only one approach for masking